std/internal/math/biguintcore.d

```diff
@@ -1549,6 +1560,70 @@ char [] biguintToHex(char [] buff, const BigDigit [] data, char separator=0,
     return buff;
 }
 
+/**
+ * Convert a big uint into an octal string.
+ *
+ * Params:
+ *  buff = The destination buffer for the octal string. Must be large enough to
+ *      store the result, including leading zeroes, which is
+ *      ceil(data.length * BigDigitBits / 3) characters.
+ *      The buffer is filled from back to front, starting from `buff[$-1]`.
+ *  data = The biguint to be converted.
+ *
+ * Returns: The index of the leading non-zero digit in `buff`. Will be
+ * `buff.length - 1` if the entire big uint is zero.
+ */
+size_t biguintToOctal(char[] buff, const(BigDigit)[] data)
+    pure nothrow @safe @nogc
+{
+    ubyte carry = 0;
+    int shift = 0;
+    size_t penPos = buff.length - 1;
+    size_t lastNonZero = buff.length - 1;
+
+    pragma(inline) void output(int digit) @nogc nothrow
```

Comment:
You're right, this should not accept a signed integer type. I refrained from using `size_t` because the range of parameters should always be < 8 (otherwise there is a bug). No need to use 64 bits when 32 bits already suffices (in fact, just 4 bits already suffice, but requiring anything smaller than 32 bits requires ugly casts and doesn't necessarily make the generated asm any better).